src/application/common/logger.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Logger from 'bunyan';
+
+const ILogger = Symbol("ILogger");
+
+interface ILogger {
```

Comment:
I like:
- The clean separation of interface/implementation
- Proxy logic was fun but I did not know about the JSON-RPC limitations, so it doesn't make sense now, so going without it is fine.

I think we should go without params: any[] like you propose for the log server function, and resolved that before sending it over to the server.

I'll rework it as such and send a v2